### ISSUE TITLE
Props are not serialized correctly using default Netwonsoft serializer

### ISSUE THREAD

**Issue by Horusiath** [akkadotnet/akka.net#569]
Netwonsoft JSON.NET is not able to correctly serialize/deserialize `Props` objects. Test code presenting problem:

```
class Creator: UntypedActor
{
    protected override void OnReceive(object message)
    {
        var props = message as Props;
        var aref = Context.ActorOf(props);
        Sender.Tell(aref);
    }
}

var creator = ActorOf<Creator>();
creator.Tell(Props.Create(() => new BlackHoleActor()));
```

In following example props message received inside Creator (message serialization is turned on) is incorrectly initialized - ActorType is set to `ActorBase` and producer to `DefaultProducer`.

**Comment by Aaronontheweb** [comment]
IMHO, related to #547 

Need strong equality by value semantics for the entirety of the `Props` class so this stuff *fails correctly* during its serialization unit tests.

**Comment by rogeralsing** [comment]
Im taking this one

**Comment by rogeralsing** [comment]
This is going to be a beast, props drags along all of the other primitives like router config, config, supervisorstrategy etc. all of them need correct serialization.

**Comment by Aaronontheweb** [comment]
Resolved as of #581 

Will be creating a new issue to deal with `Props` serialization verification.
